pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,56 @@
+package bmc
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool) error {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint: address,
+		Username: user,
+		Password: password,
+		Insecure: insecure,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		return errors.Wrap(err, "could not connect to redfish BMC.")
+	}
+	defer client.Logout()
+
+	service := client.Service
+
+	managers, err := service.Managers()
```

Comment:
You don't need all managers, you need managers that manage the given System: https://opendev.org/openstack/sushy/src/branch/master/sushy/resources/system/system.py#L413-L414

Reply:
In all tests I did this will only return a list with one Manager for the BMC.

Reply:
sushy-tools will return you one per system (which is silly, but it's the only sane way we can make virtual media work on a multi-system service).

Reply:
I was able to test an approach to retrieve the `ManagedBy` (https://github.com/stmcginnis/gofish/blob/44e5856acf6ec8e7aecd76387341248ad0288290/redfish/computersystem.go#L807).
```
 "ManagedBy":[ {"@odata.id":"/redfish/v1/Managers/iDRAC.Embedded.1"}]
```
With the information I can access the specific Manager
```
systems, _ := service.Systems()
for _, system := range systems {
    manager, err := redfish.GetManager(c, system.ManagedBy[0])
}
```
This would be ok or we need to check something more?

Reply:
I think ironic loops over system managers, we should use the same logic here (yes, we're now in the business of replicating the ironic internals. sigh.)

Reply:
Done